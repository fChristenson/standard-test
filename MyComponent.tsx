import React from "react";

export const MyComponent: React.FC = () => {
  React.useEffect(() => {
    const logFoobar = () => console.log("foobar");
    window.addEventListener("scroll", logFoobar);

    return () => {
      window.removeEventListener("scroll", logFoobar);
    };
  }, []);

  return <h1>Hello world!</h1>;
};
