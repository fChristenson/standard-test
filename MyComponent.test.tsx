import React from "react";
import { toHaveNoViolations, axe } from "jest-axe";
import { render } from "@testing-library/react";
import { MyComponent } from "./MyComponent";

expect.extend(toHaveNoViolations);

describe("MyComponent", () => {
  it("renders", () => {
    const { container } = render(<MyComponent />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("passes a11y checks", async () => {
    const { container } = render(<MyComponent />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
