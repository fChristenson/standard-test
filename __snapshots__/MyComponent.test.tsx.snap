// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent renders 1`] = `
<h1>
  Hello world!
</h1>
`;
